pkg/registry/template/etcd/etcd_test.go

```diff
@@ -0,0 +1,130 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package etcd
+
+import (
+	"testing"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/apis/extensions"
+	"k8s.io/kubernetes/pkg/fields"
+	"k8s.io/kubernetes/pkg/labels"
+	"k8s.io/kubernetes/pkg/registry/generic"
+	"k8s.io/kubernetes/pkg/registry/registrytest"
+	"k8s.io/kubernetes/pkg/runtime"
+	etcdtesting "k8s.io/kubernetes/pkg/storage/etcd/testing"
+)
+
+var namespace = "foo-namespace"
+var name = "foo-template"
+
+
+func newRest(t *testing.T) (*REST, *ProcessREST, *etcdtesting.EtcdTestServer) {
+	etcdStorage, server := registrytest.NewEtcdStorage(t, extensions.GroupName)
+	restOptions := generic.RESTOptions{Storage: etcdStorage, Decorator: generic.UndecoratedStorage, DeleteCollectionWorkers: 1}
+	rest, processRest := NewREST(restOptions)
+	return rest, processRest, server
+}
+
+func validNewTemplate() *extensions.Template {
+	return &extensions.Template{
+		ObjectMeta: api.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Spec: extensions.TemplateSpec{
+			Items: []runtime.RawExtension{},
+			Parameters: []extensions.Parameter{},
+			ObjectLabels: map[string]string{},
+		},
+	}
+}
+
+func TestCreate(t *testing.T) {
+	rest, _, server := newRest(t)
+	defer server.Terminate(t)
+	test := registrytest.New(t, rest.Store)
+	template := validNewTemplate()
+	template.ObjectMeta = api.ObjectMeta{}
+	test.TestCreate(
+		// valid
+		template,
+		// TODO: Add these cases as they exist
+	)
+}
+
+func TestUpdate(t *testing.T) {
+	rest, _, server := newRest(t)
+	defer server.Terminate(t)
+	test := registrytest.New(t, rest.Store)
+	test.TestUpdate(
+		// valid object
+		validNewTemplate(),
+		// valid update
+		func(obj runtime.Object) runtime.Object {
+			object := obj.(*extensions.Template)
+			object.Spec.ObjectLabels = map[string]string{"c": "d"}
+			return object
+		},
```

Comment:
Suggest add several invalid updates here. However since template validation isn't implemented yet, add TODO here and mention this in validation.go also.